docs/standard/glossary.md

```diff
@@ -0,0 +1,265 @@
+
+See [ASP.NET](/aspnet/#pivot=aspnet).
+
+## ASP.NET Core
+
+A cross-platform, high-performance, open source implementation of ASP.NET built on .NET Core.
+
+See [ASP.NET Core](/aspnet/#pivot=core).
+
+## CLR
+
+Common language runtime.
+
+The exact meaning depends on the context, but this usually refers to the runtime of the .NET Framework. The CLR is a virtual machine that not only executes apps but also generates and compiles code on-the-fly using a JIT compiler. The current Microsoft CLR implementation is Windows only.
+
+## CoreCLR
+
+.NET Core common language runtime.
+
+This CLR is built from the same code base as the CLR. Originally, CoreCLR was the runtime of Silverlight and was designed to run on multiple platforms, specifically Windows and OS X. CoreCLR is now part of .NET Core and represents a simplified version of the CLR. It's still a cross platform runtime. CoreCLR is also a virtual machine with JIT and code execution capabilities.
+
+## CoreFX
+
+.NET Core framework.
+
+Also known as the .NET Core Base Class Library (BCL). A set of  System.* (and to a limited extent  Microsoft.\*) libraries. The BCL is a general purpose, lower-level frarmework that higher-level application frameworks, such as ASP.NET Core, build on. The source code of the .NET Core BCL is contained in the [CoreFX repository](https://github.com/dotnet/corefx). However, the majority of the .NET Core APIs are also available in the .NET Framework, so you can think of CoreFX as a fork of the .NET Framework BCL.
+
+## CoreRT
+
+.NET Core runtime.
+
+In contrast to the CLR/CoreCLR, CoreRT is not a virtual machine, which means it doesn't include the facilities to generate and run code on-the-fly because it doesn't include a [JIT](#jit). It does, however, include the [GC](#gc) and the ability for runtime type identification (RTTI) and reflection. However, its type system is designed so that metadata for reflection isn't required. This enables having an [AOT](#aot) tool chain that can link away superfluous metadata and (more importantly) identify code that the app doesn't use. CoreRT is in development.
+
+See [Intro to .NET Native and CoreRT](https://github.com/dotnet/corert/blob/master/Documentation/intro-to-corert.md)
+
+## ecosystem
+
+All of the runtime software, development tools, and community resources that are used to build and run applications for a given technology.
+
+The term ".NET ecosystem" differs from similar terms such as ".NET stack" in its inclusion of third-party apps and libraries. Following is an example in a sentence:
+
+- "The motivation behind the .NET Standard is establishing greater uniformity in the .NET ecosystem." ([.NET Standard](net-standard.md))
+
+## framework
+
+In general, a comprehensive library that facilitates development and deployment of applications that are based on a particular technology. In this general sense, ASP.NET Core and Windows Forms are examples of application frameworks.
```

Comment:
> This section needs some clarification. A framework is different from an application framework. 

The purpose is to start with a general/common meaning of framework -- it's a highly flexible word so I'm open to suggestions here. Ideally it should work in such a way that "application framework" works as a type of framework.

> And I'm not sure that target framework and TFM are necessary here; I think that they add one more confusing detail.

The issue I'm trying to address is confusion that results from use of "framework" with different modifiers that make it have different meanings.  The fact that it is used so differently is not always immediately apparent to readers. In this case I was thinking it's important to point out that "framework" in *target framework* and TFM has a special meaning.